docs/conf.py

```diff
@@ -118,6 +118,53 @@ def minvers(name):
 .. _`astropy-dev mailing list`: http://groups.google.com/group/astropy-dev
 """
 
+# Whether to create cross-references for the parameter types in the
+# Parameters, Other Parameters, Returns and Yields sections of the docstring.
+numpydoc_xref_param_type = True
+
+# Words not to cross-reference. Most likely, these are common words used in
+# parameter type descriptions that may be confused for classes of the same
+# name.
+numpydoc_xref_ignore = {
+    'type', 'optional', 'default', 'or', 'of', 'method', 'instance',
+    'subclass', "class", "object", "keyword-only", "function", "default",
+    "thereof", "like",
+    # needed in subclassing numpy
+    "Arguments", "Path",
+    # ones I can't get rid of
+    "numpy.ma.mvoid", "CacheDamaged", "KernelSizeError", "ModelsError",
+    "InvalidTabularParameters", "HDUList", "flag", "bits",
+}
+
+# Mappings to fully qualified paths (or correct ReST references) for the
+# aliases/shortcuts used when specifying the types of parameters.
+# Numpy provides some defaults
+# https://github.com/numpy/numpydoc/blob/b352cd7635f2ea7748722f410a31f937d92545cc/numpydoc/xref.py#L62-L94
+# so we only need to define Astropy-specific X-refs
+numpydoc_xref_aliases = {
+    # python
+    'file-like': ':term:`python:file-like object`',
+    'file': ':term:`python:file object`',
+    'iterator': ':term:`python:iterator`',
+    'path-like': ':term:`python:path-like object`',
+    'module': ':term:`python:module`',
+    # matplotlib
+    'color': ':term:`color`',
+    # astropy
+    'Unit-like': ':term:`Unit-like`',
+    "angular-Quantity-like": ':term:`angular-Quantity-like`',
```

Comment:
Time-like and a few of the other terms were missing because they weren't used. I will be doing another sweep of all the docs to see if some of these terms should be implemented. It can be challenging because often times the docs don't indicate that a string is accepted, but it actually is because the input is processed by class, eg ``Time``.

Reply:
similarly, many times the docs indicate ``str`` when `path-like` might be the truer option (ie `pathlib.Path` objects also work). For example ``astropy.wcs.validate`` says ``source : str``, but I think it's more general than that.

Reply:
Yes, there may be many cases like this where the type is not accurate, but each case is a special case and would need discussion which is better to avoid in this PR otherwise it will take way too long. So better to avoid changes here.